releases/releaser/releaser.go

```diff
@@ -110,44 +107,35 @@ func (r *Releaser) uploadBinaries(ctx context.Context, params PublishParameters)
 		fmt.Printf("Uploading: %q\n", key)
```

Comment:
The logic of uploadBinary was somewhat upside down.

Previously, it opened a file, started a goroutine in a group (even though there is only one goroutine), and did the S3 upload in the goroutine, then proceeded to write to the body of the upload as gzip content. Then if the S3 Upload failed, it would write the error back to the pipe, so the the writer of the body could expose the S3 error when writing. Which can swallow errors entirely if the S3 error occurs after having read the entire body.

The changes:
open a file, create the pipe and gzip encoder, encode the file in a goroutine, if it fails to copy data close the writer with error. This ensures that only errors that occur from reading and compressing the file can happen when reading from the pipe. Upload the S3 object in the main routine and return any error that may have occured uploading to s3.